com.unity.render-pipelines.universal/Runtime/NativeRenderPass.cs

```diff
@@ -0,0 +1,528 @@
+        }
+
+        internal static uint GetSubPassAttachmentIndicesCount(ScriptableRenderPass pass)
+        {
+            uint numValidAttachments = 0;
+
+            foreach (var attIdx in pass.m_InputAttachmentIndices)
+            {
+                if (attIdx >= 0)
+                    ++numValidAttachments;
+            }
+
+            return numValidAttachments;
+        }
+
+        internal static bool AreAttachmentIndicesCompatible(ScriptableRenderPass lastSubPass, ScriptableRenderPass currentSubPass)
+        {
+            uint lastSubPassAttCount = GetSubPassAttachmentIndicesCount(lastSubPass);
+            uint currentSubPassAttCount = GetSubPassAttachmentIndicesCount(currentSubPass);
+
+            if (currentSubPassAttCount > lastSubPassAttCount)
+                return false;
+
+            uint numEqualAttachments = 0;
+            for (int currPassIdx = 0; currPassIdx < currentSubPassAttCount; ++currPassIdx)
+            {
+                for (int lastPassIdx = 0; lastPassIdx < lastSubPassAttCount; ++lastPassIdx)
+                {
+                    if (currentSubPass.m_InputAttachmentIndices[currPassIdx] == lastSubPass.m_InputAttachmentIndices[lastPassIdx])
+                        numEqualAttachments++;
+                }
+            }
+
+            return (numEqualAttachments == currentSubPassAttCount);
+        }
+
+        internal static uint GetValidColorAttachmentCount(AttachmentDescriptor[] colorAttachments)
+        {
+            uint nonNullColorBuffers = 0;
+            if (colorAttachments != null)
+            {
+                foreach (var attachment in colorAttachments)
+                {
+                    if (attachment != RenderingUtils.emptyAttachment)
+                        ++nonNullColorBuffers;
+                }
+            }
+            return nonNullColorBuffers;
+        }
+
+        internal static int FindAttachmentDescriptorIndexInList(int attachmentIdx, AttachmentDescriptor attachmentDescriptor, AttachmentDescriptor[] attachmentDescriptors)
+        {
+            int existingAttachmentIndex = -1;
+            for (int i = 0; i < attachmentIdx; ++i)
+            {
+                AttachmentDescriptor att = attachmentDescriptors[i];
+
+                if (att.loadStoreTarget == attachmentDescriptor.loadStoreTarget)
+                {
+                    existingAttachmentIndex = i;
+                    break;
+                }
+            }
+
+            return existingAttachmentIndex;
+        }
+
+        internal static int GetValidPassIndexCount(int[] array)
+        {
+            for (int i = 0; i < array.Length; ++i)
+                if (array[i] == -1)
+                    return i;
+            return array.Length - 1;
+        }
+
+        internal static Hash128 CreateRenderPassHash(int width, int height, int depthID, int sample, uint hashIndex)
+        {
+            return new Hash128((uint)width * 10000 + (uint)height, (uint)depthID, (uint)sample, hashIndex);
```

Comment:
Let's use something a bit more explicit than the multiply by 10000 as we discussed in the call. Probably a bit shift would be more intuitive.